client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,317 @@
+import { $, $a, removeFromDOM, asArray, escapeID } from '../domhelpers.js';
+import { Widget } from './widget.js';
+
+class Scoreboard extends Widget {
+  constructor(object, surface) {
+    super(object, surface);
+
+    this.addDefaults({
+      movable: true,
+      width: 300,
+      height: 200,
+      layer: -1,
+      typeClasses: 'widget scoreboard',
+      playersInColumns: true,
+      rounds: null,
+      roundLabel: 'Round',
+      scoreProperty: 'score',
+      seats: null,
+      showAllRounds: false,
+      showAllSeats: false,
+      showPlayerColors: true,
+      showTotals: true,
+      sortField: 'index',
+      sortAscending: true,
+      currentRound: null,
+      autosizeColumns: true
+    });
+  }
+
+  applyDeltaToDOM(delta) {
+    if(delta) // Don't call super unless this is a real delta to the scoreboard rather than from a seat
+      super.applyDeltaToDOM(delta);
+    this.tableCreate(this.getIncludedSeats())
+  }
+
+  classes() {
+    let className = super.classes();
+
+    if(this.get('autosizeColumns') != '')
+      className += ' equalWidth';
+
+    return className;
+  }
+
+  classesProperties() {
+    const p = super.classesProperties();
+    p.push('autosizeColumns');
+    return p;
+  }
+
+  get(property) {
+    if(property != '_totals')
+      return super.get(property)
+    else if(!this.get('showTotals'))
```

Comment:
Seems strange that _totals only works if you set showTotals. What if a game doesn't want to always show the current totals but needs to use it in calculations once in a while (e.g. the player with the highest score gets X).

Reply:
@96LawDawg What do you think? I originally had this in there because there was no requirement that scores always be numbers, so totalling them didn't always make sense. But Robert makes a good point, and I suppose that if the scores to be totalled are not numbers, we just use 0.

Reply:
@flackr's idea makes sense to me.  And as you say, do a check for number and if not a number, treat as 0.  I don't see how that's a bad idea and it has merit.  I say do it.

Reply:
Done.